test/v3/controllers/requestParsers/validators/CreateBFLossValidatorSpec.scala

```diff
@@ -14,199 +14,175 @@
-//        }
-//      }
-//    }
-//
-//    "return SelfEmploymentIdValidation error" when {
-//      "an invalid id is submitted" in {
-//        validator.validate(requestData.CreateBFLossRawData(validNino, AnyContentAsJson(createRequestBodyJson(businessId = "invalid")))) shouldBe
-//          List(BusinessIdFormatError)
-//      }
-//    }
-//
-//    "return AmountValidation error" when {
-//      "an invalid amount is submitted" in {
-//        validator.validate(
-//          requestData.CreateBFLossRawData(validNino, AnyContentAsJson(createRequestBodyJson(lossAmount = BigDecimal(100000000000.00))))) shouldBe
-//          List(RuleInvalidLossAmount)
-//      }
-//    }
-//
-//    "return BusinessIdFormatError error" when {
-//      "an invalid business id is provided" in {
-//        validator.validate(
-//          requestData.CreateBFLossRawData(validNino,
-//            AnyContentAsJson(createRequestBodyJson(businessId = "invalid")))) shouldBe
-//          List(BusinessIdFormatError)
-//      }
-//    }
-//
-//    "return multiple errors" when {
-//      "request supplied has multiple errors" in {
-//        validator.validate(
-//          requestData.CreateBFLossRawData(
-//            validNino,
-//            AnyContentAsJson(createRequestBodyJson(typeOfLoss = "self-employment-class4", businessId = "wrong", taxYear = "2010-11")))) shouldBe
-//          List(RuleTaxYearNotSupportedError, BusinessIdFormatError)
-//      }
-//    }
-//  }
-//}
+package v3.controllers.requestParsers.validators
+
+import com.typesafe.config.ConfigFactory
+import config.AppConfig
+import mocks.MockAppConfig
+import org.joda.time.DateTime
+import org.joda.time.format.{DateTimeFormat, DateTimeFormatter}
+import play.api.Configuration
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.AnyContentAsJson
+import support.UnitSpec
+import utils.CurrentDateTime
+import v3.mocks.MockCurrentDateTime
+import v3.models.errors._
+import v3.models.requestData
+import v3.models.requestData.CreateBFLossRawData
+
+class CreateBFLossValidatorSpec extends UnitSpec {
+
+  private val validNino = "AA123456A"
+  private val validTaxYear = "2018-19"
+  private val validTypeOfLoss = "self-employment"
+  private val validBusinessId = "XAIS01234567890"
+
+  val emptyBody: JsValue = Json.parse(
+    s"""{}""".stripMargin
+  )
+
+  def createRequestBodyJson(typeOfLoss: String = validTypeOfLoss,
+                            businessId: String = validBusinessId,
+                            taxYearBroughtForwardFrom: String = validTaxYear,
+                            lossAmount: BigDecimal = 1000): JsValue = Json.parse(
+    s"""{
+       |  "typeOfLoss" : "$typeOfLoss",
+       |  "businessId" : "$businessId",
+       |  "taxYearBroughtForwardFrom" : "$taxYearBroughtForwardFrom",
+       |  "lossAmount" : $lossAmount
+       |}""".stripMargin
+  )
+
+  class Test(errorFeatureSwitch: Boolean = true) extends MockCurrentDateTime with MockAppConfig {
```

Comment:
Unless we are needing to feature switch this particular error... 
```suggestion
  class Test extends MockCurrentDateTime {
```